File: Interpreter/Golang/OrExpression.go
package main

import "strings"

type OrExpression struct {
	expression1 IExpression
	expression2 IExpression
}

func (orExpression *OrExpression) solve(context string) bool {
	if !strings.Contains(context, "||") {
		return false
	}

	return orExpression.expression1.solve(context) || orExpression.expression2.solve(context)
} 

File: Interpreter/Dart/OrExpression.dart
import 'IExpression.dart';

class OrExpression implements IExpression {
  IExpression? _expression1, _expression2;

  OrExpression(this._expression1, this._expression2);

  @override
  bool solve(String context) {
    if (!context.contains("||")) {
      return false;
    }

    return _expression1!.solve(context) || _expression2!.solve(context);
  }
}

